python/dcr_template.py

```diff
@@ -39,61 +40,132 @@
 import lsst.meas.algorithms as measAlg
 import lsst.pex.exceptions
 import lsst.pex.policy as pexPolicy
-from lsst.sims.photUtils import Bandpass, PhotometricParameters
+from lsst.sims.photUtils import Bandpass
 from lsst.utils import getPackageDir
 
 from .calc_refractive_index import diff_refraction
 
-__all__ = ["DcrModel", "DcrCorrection"]
+
+__all__ = ("DcrModel", "DcrCorrection")
 
 nanFloat = float("nan")
 nanAngle = Angle(nanFloat)
 lsst_lat = Angle(np.radians(-30.244639))
 lsst_lon = Angle(np.radians(-70.749417))
 lsst_alt = 2663.
 
+# Temporary debugging parameters, used if debug_mode=True or self.debug=True is set.
+x0 = 300
+dx = 200
+y0 = 500
+dy = 200
+
 
 class DcrModel:
-    """!Lightweight object with only the minimum needed to generate DCR-matched template exposures."""
+    """!Lightweight object with only the minimum needed to generate DCR-matched template exposures.
+
+    A model must first be generated with DcrCorrection (below). That model can then be used directly, or
+    persisted and later read back in needing only this more lightweight class that doesn't need to carry
+    around the input exposures. This class will generate template exposures suitable for
+    image differencing that are matched to existing exposures in a repository.
+
+    Attributes
+    ----------
+    bandpass : lsst.sims.photUtils.Bandpass object
+        Bandpass object returned by load_bandpass
+    bbox : lsst.afw.geom.Box2I object
+        A bounding box.
+    butler : lsst.daf.persistence Butler object
+        The butler handles persisting and depersisting data to and from a repository.
+    debug : bool
+        Temporary debugging option.
+        If set, calculations are performed on only a small region of the full images.
+    detected_bit : int
+        Value of the detected bit in the bit plane mask.
+    exposure_time : float
+        Length of the exposure, in seconds.
+    filter_name : str
+        Name of the bandpass-defining filter of the data. Expected values are u,g,r,i,z,y.
+    instrument : str
+        Name of the observatory. Used to format dataIds for the butler.
```

Comment:
The problem is that each camera I've used (currently LsstSim and Decam) has an entirely different set of keys for the butler, and the interface is somewhat fragile. I'm going to leave this as-is for now, but I've filed ticket DM-9851 to improve this.